crates/ruff_python_ast/src/whitespace.rs

```diff
@@ -1,15 +1,21 @@
-use ruff_text_size::TextRange;
+use ruff_text_size::{TextRange, TextSize};
 use rustpython_parser::ast::Ranged;
 
 use crate::source_code::Locator;
 
 /// Extract the leading indentation from a line.
+#[inline]
 pub fn indentation<'a, T>(locator: &'a Locator, located: &T) -> Option<&'a str>
 where
     T: Ranged,
 {
-    let line_start = locator.line_start(located.start());
-    let indentation = &locator.contents()[TextRange::new(line_start, located.start())];
+    indentation_at_offset(locator, located.start())
```

Comment:
We should avoid generic functions with a lot of logic in them. Rust unrolls them for each generic type. The solution is to extract the non-generic part and call into it.